packages/zowe-explorer-api/src/vscode/ui/utils/TableBuilder.ts

```diff
@@ -0,0 +1,224 @@
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import { ExtensionContext } from "vscode";
+import { Table } from "../TableView";
+import { TableMediator } from "./TableMediator";
+
+/**
+ * A builder class for quickly instantiating {@link Table.View} instances.
+ * Especially useful for building multiple tables with similar layouts or data.
+ *
+ * @remarks
+ *
+ * ## Building a table
+ *
+ * Developers can build a table using the helper methods provided by the builder.
+ * Once a table is ready to be built, use the {@link TableBuilder.build} function to create
+ * a new {@link Table.View} instance with the given configuration.
+ *
+ * ## Sharing tables
+ *
+ * Share a table during the build process by using the {@link TableBuilder.buildAndShare} function.
+ * This function will create the {@link Table.View} instance while also adding it to the mediator
+ * in Zowe Explorer's extender API. Extenders who would like to contribute to shared tables can do
+ * so by accessing the table by its unique ID.
+ */
+export class TableBuilder {
+    private context: ExtensionContext;
+    private data: Table.ViewOpts = {
+        actions: {
+            all: [],
+        },
+        contextOpts: {
+            all: [],
+        },
+        columns: [],
+        rows: [],
+        title: "",
+    };
+    private forWebviewView = false;
+
+    public constructor(context: ExtensionContext) {
+        this.context = context;
+    }
+
+    public isView(): this {
+        this.forWebviewView = true;
+        return this;
+    }
+
+    /**
+     * Set optional properties for the table view.
+     * @param opts The options for the table
+     * @returns The same {@link TableBuilder} instance with the options added
+     */
+    public options(opts: Table.GridProperties): this {
+        this.data = { ...this.data, options: this.data.options ? { ...this.data.options, ...opts } : opts };
+        return this;
+    }
+
+    /**
+     * Set the title for the next table.
+     * @param name The name of the table
+     * @returns The same {@link TableBuilder} instance with the title added
+     */
+    public title(name: string): this {
+        this.data.title = name;
+        return this;
+    }
+
+    /**
+     * Set the rows for the next table.
+     * @param rows The rows of content to use for the table
+     * @returns The same {@link TableBuilder} instance with the rows added
+     */
+    public rows(...rows: Table.RowData[]): this {
+        this.data.rows = rows;
+        return this;
+    }
+
+    /**
+     * Adds rows to the table. Does not replace existing rows.
+     * @param rows The rows of content to add to the table
+     * @returns The same {@link TableBuilder} instance with the new rows added
+     */
+    public addRows(rows: Table.RowData[]): this {
+        this.data.rows = [...this.data.rows, ...rows];
+        return this;
```

Comment:
I like that most (if not all) of these functions return `this`.
It helps to concatenate methods calls much faster.
I'm wondering if we should do more of `this` going forward 😋

Reply:
Thanks! I figured the builder pattern would be useful here, but there might be a few other places in the codebase where this could be helpful 😁